src/librustc/middle/stability.rs

```diff
@@ -513,23 +513,8 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                 // handled by the lint emitting logic above.
             }
             None => {
-                // This is an 'unmarked' API, which should not exist
-                // in the standard library.
-                if self.sess.features.borrow().unmarked_api {
-                    self.sess.struct_span_warn(span, "use of unmarked library feature")
-                             .span_note(span, "this is either a bug in the library you are \
-                                               using or a bug in the compiler - please \
-                                               report it in both places")
-                             .emit()
-                } else {
-                    self.sess.struct_span_err(span, "use of unmarked library feature")
-                             .span_note(span, "this is either a bug in the library you are \
-                                               using or a bug in the compiler - please \
-                                               report it in both places")
-                             .span_note(span, "use #![feature(unmarked_api)] in the \
-                                               crate attributes to override this")
-                             .emit()
-                }
+                span_bug!(span, "Encountered unmarked API, which should \
+                                 not exist in the standard library");
```

Comment:
Encountering unmarked API is not a library's  fault, but a compiler bug. Could you tweak the message to not mention the standard library, e.g. just "encountered unmarked API".